crates/primitives/src/transaction/sidecar.rs

```diff
@@ -503,3 +503,155 @@ fn generate_blob_sidecar(blobs: Vec<Blob>) -> BlobTransactionSidecar {
 
     BlobTransactionSidecar { blobs, commitments, proofs }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        hex,
+        kzg::{Blob, Bytes48},
+        transaction::sidecar::generate_blob_sidecar,
+        BlobTransactionSidecar,
+    };
+    use std::{fs, fs::File, io::Read, path::PathBuf};
+
+    #[test]
+    fn test_blob_transaction_sidecar_generation() {
+        // Read the contents of the JSON file into a string.
+        let json_content = fs::read_to_string(
+            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/transaction/blob_data/blob1.json"),
+        )
+        .expect("Failed to read the blob data file");
+
+        // Parse the JSON contents into a serde_json::Value
+        let json_value: serde_json::Value =
+            serde_json::from_str(&json_content).expect("Failed to deserialize JSON");
+
+        // Extract blob data from JSON and convert it to Blob
+        let blobs: Vec<Blob> = vec![Blob::from_hex(
+            json_value.get("data").unwrap().as_str().expect("Data is not a valid string"),
+        )
+        .unwrap()];
+
+        // Generate a BlobTransactionSidecar from the blobs
+        let sidecar = generate_blob_sidecar(blobs.clone());
+
+        // Assert commitment equality
+        // Random blob details: https://sepolia.blobscan.com/blob/0x01915bb277dc526cc966eab02e4636d655d8b5ffc81cdd51122c97b04ecd261a
+        assert_eq!(
+            sidecar.commitments,
+            vec![
+                Bytes48::from_hex(json_value.get("commitment").unwrap().as_str().unwrap()).unwrap()
+            ]
+        );
+    }
+
+    #[test]
+    fn test_blob_transaction_sidecar_size() {
+        // Vector to store blob data from each file
+        let mut blobs: Vec<Blob> = Vec::new();
+
+        // Iterate over each file in the folder
+        for entry in fs::read_dir(
+            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/transaction/blob_data/"),
+        )
+        .expect("Failed to read blob_data folder")
+        {
+            let entry = entry.expect("Failed to read directory entry");
+            let file_path = entry.path();
+
+            // Ensure the entry is a file and not a directory
+            if file_path.is_file() {
```

Comment:
flip this to continue if not file to get rid of nested scope

Reply:
done